From 288aee36a46fe6f4d3037bd3f0c11241ab143243 Mon Sep 17 00:00:00 2001
From: Dinesh Kumar Mirche <dmirche@habana.ai>
Date: Tue, 13 May 2025 06:58:42 +0300
Subject: [PATCH] [SW-222126] gaudi3: sec: avoid reusing range registers

This patch avoid the reusing the same range register
in preboot and arcmgmt. This might block different
ranges instead of expected ranges and will have security
violations which are not expected.

Change-Id: I29f3606a338eb7e5df06729b0526a03f51714c44
Signed-off-by: Dinesh Kumar Mirche <dmirche@habana.ai>
---

diff --git a/drivers/misc/hl_security/gaudi3/security.c b/drivers/misc/hl_security/gaudi3/security.c
index eeddefc..45d9449 100644
--- a/drivers/misc/hl_security/gaudi3/security.c
+++ b/drivers/misc/hl_security/gaudi3/security.c
@@ -791,18 +791,18 @@
 	 * Start: HD[0..7]_TPC[0..7]_EML_TPC_QM + 4KB (e.g. 0x300007FF004E000 for HD0_TPC0)
 	 * End: HD[0..7]_TPC[0..7]_EML_CS - 1 (e.g. 300007FF01FEFFF for HD0_TPC0)
 	 */
-	/* H9_PROJECT ➔ D0_PCIE_MSTR_RR_MSTR_IF ➔ RR_LBW ➔ PRIV_RANGE_MASK_SHORT_9
+	/* H9_PROJECT ➔ D0_PCIE_MSTR_RR_MSTR_IF ➔ RR_LBW ➔ PRIV_RANGE_MASK_SHORT_11
 	 * used to set MASK to cover TPC_DBG gap regions for HD[0..7] TPC[0..7]
 	 */
 	val = CFG_RREG32(mmD0_PCIE_MSTR_RR_MSTR_IF_RR_LBW_BASE +
-			 mmRTR_CTRL_RR_LBW_PRIV_RANGE_MASK_SHORT_9);
+			 mmRTR_CTRL_RR_LBW_PRIV_RANGE_MASK_SHORT_11);
 	val &= ~(TPC_DBG_GAP_MASK);
 	CFG_WREG32((mmD0_PCIE_MSTR_RR_MSTR_IF_RR_LBW_BASE +
-		    mmRTR_CTRL_RR_LBW_PRIV_RANGE_MASK_SHORT_9),
+		    mmRTR_CTRL_RR_LBW_PRIV_RANGE_MASK_SHORT_11),
 		   val);
 	min = TPC_DBG_BASE + mmHD0_TPC0_EML_TPC_QM_BASE + HD0_TPC0_EML_TPC_QM_MAX_OFFSET;
 	max = TPC_DBG_BASE + mmHD0_TPC0_EML_CS_BASE - 1;
-	hl_secure_lbw_rr(dev, HL_SEC_LBW_RR_SHORT_9, min, max, rr_sel, true, HL_SEC_RD_WR_PROT);
+	hl_secure_lbw_rr(dev, HL_SEC_LBW_RR_SHORT_11, min, max, rr_sel, true, HL_SEC_RD_WR_PROT);
 
 	/* All PCIE type RRs(PCIE_ELBI, PCIE_MSTR, PCIE_LBW) for all RR instances are not configured
 	 * during security configurations. But clearing for all  should not have any adverse impact.
@@ -816,6 +816,7 @@
 	hl_secure_lbw_rr(dev, HL_SEC_LBW_RR_SHORT_1, min, max, rr_sel, false, HL_SEC_RD_WR_PROT);
 	hl_secure_lbw_rr(dev, HL_SEC_LBW_RR_SHORT_2, min, max, rr_sel, false, HL_SEC_RD_WR_PROT);
 	hl_secure_lbw_rr(dev, HL_SEC_LBW_RR_SHORT_8, min, max, rr_sel, false, HL_SEC_RD_WR_PROT);
+	hl_secure_lbw_rr(dev, HL_SEC_LBW_RR_SHORT_9, min, max, rr_sel, false, HL_SEC_RD_WR_PROT);
 	hl_secure_lbw_rr(dev, HL_SEC_LBW_RR_SHORT_10, min, max, rr_sel, false, HL_SEC_RD_WR_PROT);
 	hl_secure_lbw_rr(dev, HL_SEC_LBW_RR_SHORT_13, min, max, rr_sel, false, HL_SEC_RD_WR_PROT);
 }
